purchase_package_qty/models/purchase_order_line.py

```diff
@@ -22,10 +24,9 @@
 ##############################################################################
 
 from math import ceil
-
-from openerp.osv.osv import except_osv
-from openerp import api, fields, models, _
+from odoo import api, fields, models, _
 import openerp.addons.decimal_precision as dp
+from odoo.exceptions import Warning
```

Comment:
Don't forget `Warning` is deprecated.
We should replace for `UserError`.

No need to fix it now, it's not blocking. But please whenever you are tempted to use `Warning`, import `UserError` instead ;)